content/blog/build-your-own-ilo-redfish-simulator.md

```diff
@@ -0,0 +1,146 @@
+
+For POST, PUT and PATCH requests, the simulator performs limited verification of the query, modifies the requested endpoint and sends back a `204 No Content` status code with no associated body. For the same request, a real Redfish service performs additional verification and sends back a non-empty response body with a `20X` status code. 
+
+The different behavior of the simulator, compared to a real iLO 5 Redfish service can be illustrated with a PATCH request for modifying the `IndicatorLED` resource of a computer chassis. The authorized values for this parameter are defined by the DMTF in the Chassis Redfish schema. For an iLO 5 with firmware version 2.30, the [implemented schema](http://redfish.dmtf.org/schemas/v1/Chassis.v1_10_2.json#/definitions/IndicatorLED) specifies the following possible values: `unknown`, `Lit`, `Blinking`, `Off`. A physical iLO 5 complains if you supply a value different from what the schema proposes. However, the DMTF Redfish Mockup Server accepts any string as shown in the following screenshot. 
+
+The code of this picture sends a `PATCH` request with and invalid value (`Foo`) toward an iLO 5 simulator. The simulator performs the patch action and responds with status code `204`. The last command of shows that the action has been successfully performed.
+
+![PATCH an invalid property in the DMTF Redfish Mockup Server](/img/fooindicatorled.png "PATCH an invalid property in the DMTF Redfish Mockup Server")
+
+The same query against a physical iLO 5 returns a `400 Bad Request` status and a `@Message.ExtendedInfo` mentioning the faulty argument.
+
+![PATCH of an invalid property in a physical iLO 5](/img/fooindicatorledagainstilo5.png "PATCH of an invalid property in a physical iLO 5")
+
+### Querying the Mockup Server with iLOrest
+
+If you want to query your iLO simulator with [iLOrest](http://hpe.com/info/resttool) you have to extract the cache directory you saved during the mockup creation (see the "Preparing the use of iLOrest against your Mockup Server" paragraph above) and edit the `url` property of its two files: `index` and `<longUniqIdenfier>` to make them point to the simulator.
+
+On a Linux system, this operation can be done with the following commands:
+
+```Shell
+cd $iloCacheDir/cache/
+sed -i 's?\("url": "https://\)ilo-IP-physical"?\1ilo-IP-simulator"?' *
+```
+Once the iLOrest cache points to your mockup server, you can use this Redfish client tool to query the mockup. 
+
+> **NOTE** If you use the `login` iLOrest commands, the cache will be overwritten. If you use the `logout` command, the cache will be erased.
+
+## Conclusion
+
+To learn and test the Redfish API, The DMTF provides two very useful tools to create Redfish mockups and simulate a live service. They are easy to install and use and their quality is a good. In addition, from my personal experience, I found the maintainers of these active GithHub projects very responsive to address quality issues and proposed enhancements. 
```

Comment:
"The DMTF" should be "the DMTF"
Place a comma after "They are easy to install and use" in order to set it apart from the next thought, which is "and their quality is good."
Also, remove the "a" before "good".
"GithHub" should probably be "GitHub"